src/protocols/hpr_protocol_router.erl

```diff
@@ -1,12 +1,89 @@
 -module(hpr_protocol_router).
 
+-include("grpc/autogen/server/packet_router_pb.hrl").
+-include("grpc/autogen/client/router_pb.hrl").
+
+% TODO: should be in a common include file
+-define(JOIN_REQUEST, 2#000).
+
 -export([send/3]).
 
 -spec send(
     Packet :: hpr_packet_up:packet(),
     HandlerPid :: pid(),
-    Routes :: hpr_route:route()
+    Route :: hpr_route:route()
 ) -> ok | {error, any()}.
-send(_Packet, _HandlerPid, _Route) ->
-    %% TODO: Do unary call the GRPC router service here
-    ok.
+send(Packet, _HandlerPid, Route) ->
+    StateChannelMsg = blockchain_state_channel_message_v1(Route, Packet),
+    wrap_grpc_response(helium_router_client:route(StateChannelMsg)).
+
+% translate hpr_packet_up:packet into blockchain_state_channel_message_v1
+-spec blockchain_state_channel_message_v1(
+    hpr_route:route(), hpr_packet_up:packet()
+) ->
+    router_pb:blockchain_state_channel_message_v1_pb().
+blockchain_state_channel_message_v1(Route, HprPacketUp) ->
+    % Decompose uplink message
+    #packet_router_packet_up_v1_pb{
```

Comment:
We have a `hpr_packet_up:new/1` it's only exported for test but you can probably move that out and use it here to avoid using record directly

Reply:
Is there a reason to avoid using the record directly? Our code is kind of a mishmash of using the functions and record directly.